scylla-cdc/src/stream_generations.rs

```diff
@@ -96,11 +95,13 @@ impl GenerationFetcher {
 
         let result = self
             .session
-            .query(query, (value::CqlTimestamp(time.num_milliseconds()),))
+            .query_unpaged(query, (value::CqlTimestamp(time.num_milliseconds()),))
             .await?
-            .rows;
+            .into_rows_result()?
+            .maybe_first_row::<(GenerationTimestamp,)>()?
+            .map(|(ts,)| ts);
 
-        GenerationFetcher::return_single_row(result)
+        Ok(result)
     }
```

Comment:
:thinking: This returns at most one row, but could possibly scan through many tombstones. @Lorak-mmk WDYT?

Reply:
Again, no idea. cc: @piodul

Reply:
It's fine, the table being scanned here is not big and rarely changes (only during topology changes). The tombstones should expire quickly enough that they do not cause performance problems.